Vcctl panic while listing queues after queue was deleted but podgroups remain

### Description

When a queue is removed, but there is still some podgroups belonging to this queue. Using `vcctl queue list` command for querying queues will cause a panic.

### Steps to reproduce the issue

1. Create a queue
```
$ kubectl get q 
NAME      AGE
default   98d
root         56d
test          5s
```
2. Create some podgroups belonging to this queue
```
$ kubectl get pg -owide
NAME                                                                                         STATUS    MINMEMBER   RUNNINGS   AGE   QUEUE
test1-ea5776cd-6337-4a0e-b0cf-11570828baf7      Running   1                          1                       10s     test
test2-3cd12bbe-59b1-4345-a1f4-716d451bd3ad   Running   1                                                    3s       test
```
3. Delete the queue
```
$ kubectl delete q test
queue.scheduling.volcano.sh "test" deleted
```
4. use `vcctl queue list` to query queues
```
$ vcctl queue list
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x15d4909]

goroutine 1 [running]:
volcano.sh/volcano/pkg/cli/podgroup.(*PodGroupStatistics).StatPodGroupCountsForQueue(...)
        /volcano/pkg/cli/podgroup/podgroup.go:20
volcano.sh/volcano/pkg/cli/queue.ListQueue({0x1bf90c0, 0x291c560})
        /volcano/pkg/cli/queue/list.go:107 +0x489
main.buildQueueCmd.func4(0xc00027ef08, {0x1960e18?, 0x4?, 0x1960e1c?})
        /volcano/cmd/cli/queue.go:66 +0x29
github.com/spf13/cobra.(*Command).execute(0xc00027ef08, {0x291c560, 0x0, 0x0})
        /go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:989 +0xa91
github.com/spf13/cobra.(*Command).ExecuteC(0xc00032e308)
       /go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
       /go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
k8s.io/component-base/cli.run(0xc00032e308)
        /go/pkg/mod/k8s.io/component-base@v0.32.2/cli/run.go:143 +0x245
k8s.io/component-base/cli.Run(0xc00032e308?)
        /go/pkg/mod/k8s.io/component-base@v0.32.2/cli/run.go:44 +0x17
main.main()
        /volcano/cmd/cli/vcctl.go:43 +0x1c5

```


### Describe the results you received and expected

After the queue was deleted, `vcctl queue list` command will ignore the podgroups whose queues are deleted, and won't panic.

### What version of Volcano are you using?

master

### Any other relevant information

_No response_

--- a/pkg/cli/queue/list.go
+++ b/pkg/cli/queue/list.go
@@ -104,6 +104,10 @@ func ListQueue(ctx context.Context) error {
 	}
 
 	for _, pg := range pgList.Items {
+		_, ok := queueStats[pg.Spec.Queue]
+		if !ok {
+			continue
+		}
 		queueStats[pg.Spec.Queue].StatPodGroupCountsForQueue(&pg)
 	}
 

--- a/pkg/cli/queue/queue_test.go
+++ b/pkg/cli/queue/queue_test.go
@@ -21,8 +21,11 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"io"
 	"net/http"
 	"net/http/httptest"
+	"os"
+	"strings"
 	"testing"
 
 	"github.com/spf13/cobra"
@@ -247,5 +250,103 @@ test-queue               1       Open    root    1       2       3       4      
 				t.Errorf("expected %q, got %q", tc.expected, got)
 			}
 		})
+	}
+}
+
+// Test ListQueue can output results normally when there are residual podgroups with their bound queues not exist.
+func TestListQueue_residualPg(t *testing.T) {
+	mockServer := func() *httptest.Server {
+		queues := v1beta1.QueueList{}
+		queues.Items = []v1beta1.Queue{
+			{
+				ObjectMeta: v1.ObjectMeta{
+					Name: "testQueue1",
+				},
+			},
+			{
+				ObjectMeta: v1.ObjectMeta{
+					Name: "testQueue2",
+				},
+			},
+		}
+		podGroups := v1beta1.PodGroupList{}
+		podGroups.Items = []v1beta1.PodGroup{
+			{
+				ObjectMeta: v1.ObjectMeta{
+					Name: "testPodGroup1",
+				},
+				Spec: v1beta1.PodGroupSpec{
+					Queue: "testQueue1",
+				},
+			},
+			{
+				ObjectMeta: v1.ObjectMeta{
+					Name: "testPodGroup2",
+				},
+				Spec: v1beta1.PodGroupSpec{
+					Queue: "testQueue2",
+				},
+			},
+			{
+				ObjectMeta: v1.ObjectMeta{
+					Name: "testPodGroup3",
+				},
+				Spec: v1beta1.PodGroupSpec{
+					Queue: "testQueue3",
+				},
+			},
+		}
+		handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("Content-Type", "application/json")
+			if strings.HasSuffix(r.URL.Path, "queues") {
+				val, err := json.Marshal(queues)
+				if err == nil {
+					w.Write(val)
+				}
+			} else if strings.HasSuffix(r.URL.Path, "podgroups") {
+				val, err := json.Marshal(podGroups)
+				if err == nil {
+					w.Write(val)
+				}
+			}
+		})
+		return httptest.NewServer(handler)
+	}
+
+	InitListFlags(&cobra.Command{})
+	server := mockServer()
+	defer server.Close()
+
+	// Capture stdout
+	originStdout := os.Stdout
+	r, w, err := os.Pipe()
+	if err != nil {
+		t.Fatalf("Create pipe failed: %v", err)
+	}
+	os.Stdout = w
+	defer func() {
+		os.Stdout = originStdout
+		w.Close()
+	}()
+
+	listQueueFlags.CommonFlags = getCommonFlags(server.URL)
+
+	expectOutput := `Name                     Weight  State   Parent  Inqueue Pending Running Unknown Completed
+testQueue1               0                       0       0       0       0       0       
+testQueue2               0                       0       0       0       0       0`
+	err = ListQueue(context.TODO())
+	if err != nil {
+		t.Errorf("List queue failed: %v", err)
+		return
+	}
+	w.Close()
+	out, err := io.ReadAll(r)
+	if err != nil {
+		t.Fatalf("Read list queue output failed: %v", err)
+	}
+	result := strings.TrimSpace(string(out))
+
+	if expectOutput != result {
+		t.Errorf("(Test list queues with residual podgroups failed): \nExpect:\n%vGot:\n%v", expectOutput, result)
 	}
 }

